loopout.go

```diff
@@ -900,29 +914,47 @@ func (s *loopOutSwap) sweep(ctx context.Context,
 		return s.htlc.GenSuccessWitness(sig, s.Preimage)
 	}
 
+	var (
+		remainingBlocks    = s.CltvExpiry - s.height
+		blocksToLastReveal = remainingBlocks - MinLoopOutPreimageRevealDelta
+		preimageRevealed   = s.state == loopdb.StatePreimageRevealed
+	)
+
 	// Calculate the transaction fee based on the confirmation target
 	// required to sweep the HTLC before the timeout. We'll use the
 	// confirmation target provided by the client unless we've come too
 	// close to the expiration height, in which case we'll use the default
 	// if it is better than what the client provided.
 	confTarget := s.SweepConfTarget
-	if s.CltvExpiry-s.height <= DefaultSweepConfTargetDelta &&
+	if remainingBlocks <= DefaultSweepConfTargetDelta &&
 		confTarget > DefaultSweepConfTarget {
+
 		confTarget = DefaultSweepConfTarget
 	}
+
 	fee, err := s.sweeper.GetSweepFee(
 		ctx, s.htlc.AddSuccessToEstimator, s.DestAddr, confTarget,
 	)
 	if err != nil {
 		return err
 	}
 
+	// If we have not revealed our preimage, and we don't have time left
+	// to sweep the swap, we abandon the swap because we can no longer
+	// sweep the success path (without potentially having to compete with
+	// the server's timeout sweep), and we have not had any coins pulled
+	// off-chain.
+	if blocksToLastReveal <= 0 && !preimageRevealed {
```

Comment:
Ah should have added an explanation here! In a follow up PR which adds reporting swap cancellation to the server, I want to report the miner fee and estimate to the server, so I put this below so that I don't need to move this in the next PR. 

Get fee calls aren't too expensive, and we'll only have one extra call before we exit for good if this condition is hit, so I thought it was ok?